src/data/thunks.js

```diff
@@ -29,3 +29,20 @@ export function fetchCourseDetail(courseId) {
     }
   };
 }
+
+export function fetchWaffleFlags(courseId) {
+  return async (dispatch) => {
+    dispatch(updateStatus({ courseId, status: RequestStatus.IN_PROGRESS }));
+
+    try {
+      const waffleFlags = await getWaffleFlags(courseId);
+      dispatch(updateStatus({ courseId, status: RequestStatus.SUCCESSFUL }));
+      dispatch(fetchWaffleFlagsSuccess({ waffleFlags }));
+    } catch (error) {
+      // If fetching the waffle flags is unsuccessful,
+      // the pages will still be accessible and display without any issues.
+      // eslint-disable-next-line no-console
+      console.error({ courseId, status: RequestStatus.NOT_FOUND });
+    }
+  };
+}
```

Comment:
Optional: since these waffle flags are a totally new API, I would recommend you load them with React Query instead of storing them in the global redux state.

Example of similar code:

https://github.com/openedx/frontend-app-authoring/blob/841aede8cd2445a39439df06b353d6283d280ba8/src/library-authoring/data/apiHooks.ts#L116-L125

Reply:
Also, if you do it this way, and you define types on your `getWaffleFlags()` function in `api.js`, then any place you use the new `useCourseWaffleFlags()` hook, you'll have a typed response that properly validates all of the waffle flag names available in the response.

```ts
const { data: waffleFlags, isLoading: isLoadingFlags } = useCourseWaffleFlags();
data?.useNewGradingPage; // <- will have type boolean
data?.useOtherFlag; // <- will show an error
```

Reply:
This is a great idea, and I really like it! However, at this stage, we’re already managing quite a few changes in the current PR. Implementing this suggestion would require additional time for development and testing, which we can’t accommodate right now. I suggest we proceed with the current solution for now, but I would definitely consider this improvement for future iterations 💯

Reply:
OK but in the future, please always prefer using React Query and TypeScript instead of Redux and JavaScript when building new things. It's a lot simpler and nicer.

Or, ideally, you could open a follow-up refactor PR after this one merges.

Reply:
While I'm very much in favor of moving off of Redux to React Query (and Typescript), I have very mixed feelings about a codebase that is inconsistent in its stack.  It makes for a very confusing developer experience.

Which is to say, we should start talking about how we'll finish this move, since we seem to have already started it.

Anyway, as far as this PR is concerned, I'm fine with it staying on Redux.

Reply:
JS and TS can be "mixed and matched" without any problems, and without too much developer confusion.

Redux and React Context+Query is definitely another matter. But I really don't want to keep adding things to the Redux state, making it even bigger and harder to refactor, just because it's there. I'm not sure what other approach we can take besides making wholly new things in React Query and slowly transitioning other chunks of the state from Redux to Context+Query ?

Reply:
I guess the other possible approach would be to embark on a full refactor, at least on a per-repo basis.  I've started talking to people about the feasibility of doing this.

Reply:
That would be great and I'd love to support that / work on it.